dbms/src/Parsers/ASTDictionary.cpp

```diff
@@ -0,0 +1,164 @@
+#include <Parsers/ASTDictionary.h>
```

Comment:
I thought that we can parse free-form syntax, convert to DOM and re-use existing code.

Reply:
It can be better, because <source> can have custom elements depend on the source itself.

Reply:
Yes, I also plan to convert AST to DOM (Poco::AbstractConfiguration) and reuse the current code in the ExternalLoader. `SOURCE` is the only section where we have absolutely custom properties and it's already represented as [function with key-value arguments](https://github.com/ClickHouse/ClickHouse/pull/7209/files/0e664807e9029b94c3a25b2b9bd0a61626a3de5b#diff-65bc97a2f72ddc9a11333338f2a7b4efR59) which can hold any amount of custom properties (also nested). 

Or you mean some general conversion function?